code/game/machinery/machinery.dm

```diff
@@ -268,11 +268,16 @@ Class Procs:
 	if(!interact_offline && stat & (BROKEN|MAINT))
 		user << "<span class='danger'>\The [src] seems broken.</span>"
 		return 1
-	src.add_fingerprint(user)
+	add_fingerprint(user)
```

Comment:
`if(!interact_offline && stat & (BROKEN|MAINT))` should be `if(!interact_offline && (stat & (BROKEN|MAINT)))` because BYOND has weird operator precedence.